### PR TITLE
[WIP] Annotation table auto-increment column

### PR DESCRIPTION
**Pull request by marcospri**

### CODE REVIEW

File: h/models/annotation.py [hunk 1]
```diff
@@ -38,6 +38,11 @@ class Annotation(Base):
         types.URLSafeUUID, server_default=sa.func.uuid_generate_v1mc(), primary_key=True
     )
 
+    #: Annotation PK: this was added after annotation.id and it's meant to replace it short term
+    #: For now we'll have a sequence and a nullable serial column.
+    pk_sequence = sa.Sequence("annotation_id_seq", cycle=False)
+    pk = sa.Column(sa.Integer, nullable=True, unique=True)
```
Comment (marcospri) [inline h/models/annotation.py:44]: This is not the table's PK but it's call PK here.

`annotation.id` is taken by the UUID column. I reckon ideally we call that `annotation.uuid` or `annotation.pubid` but I don't think we should rename that until the new column is the pk.

We could in the interim name this `annotation.private_id` or similar but that would be inconsistent with all the other `table.id` columns.

`annotation.pk` I reckon is the best for now.

File: h/models/annotation_metadata.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import sqlalchemy as sa
+from sqlalchemy.ext.mutable import MutableDict
+
+
+class AnnotationMetadata(Base):
+    __tablename__ = "annotation_metadata"
+
+    id = sa.Column(sa.Integer, autoincrement=True, primary_key=True)
+
+    annotation_pk = sa.Column(
```
Comment (marcospri) [inline h/models/annotation_metadata.py:12]: I was more sure about `annotation.pk` in the annotation table itself.

Here I wanted to just go for `annotation_id` to future prof the name but we already have tables that relate to annotations by the UUID with a column name `annotation_id`.

File: h/models/annotation.py [hunk 3]
```diff
@@ -38,6 +38,11 @@ class Annotation(Base):
         types.URLSafeUUID, server_default=sa.func.uuid_generate_v1mc(), primary_key=True
     )
 
+    #: Annotation PK: this was added after annotation.id and it's meant to replace it short term
+    #: For now we'll have a sequence and a nullable serial column.
+    pk_sequence = sa.Sequence("annotation_id_seq", cycle=False)
```
Comment (marcospri) [inline h/models/annotation.py:43]: We don't want the column to be auto-incrementing right now, we just create the sequence for now to start filling the column with the task.

When we are done with that we can make the `server_default` of this column the sequence value like any other auto incrementing column.

File: h/security/identity.py [hunk 4]
```diff
@@ -84,6 +84,14 @@ class Identity:
     user: Optional[LongLivedUser] = None
     auth_client: Optional[LongLivedAuthClient] = None
 
+    @property
+    def is_lms(self):
```
Comment (marcospri) [inline h/security/identity.py:88]: This I found while doing looking at security. I reckon `is_lms` is going to be useful right now.

Of course this won't be in a PR about adding a new column.

File: h/tasks/annotations.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    # pylint: disable=no-member
+    with celery.request.tm:
+        db = celery.request.db
+
+        annotations = (
+            db.query(Annotation.id)
+            .where(Annotation.pk == None)
+            .order_by(Annotation.updated.asc())
+            .limit(batch_size)
+        ).cte("annotations")
```
Comment (marcospri) [inline h/tasks/annotations.py:19]: I haven't found another way to do the sorting and updating.

I reckon this reads nicely.

File: tox.ini [hunk 6]
```diff
@@ -83,6 +83,7 @@ setenv =
     dev: ENABLE_WEB = {env:ENABLE_WEB:true}
     dev: ENABLE_WEBSOCKET = {env:ENABLE_WEBSOCKET:true}
     dev: ENABLE_WORKER = {env:ENABLE_WORKER:true}
+    dev: ALEMBIC_CONFIG = {env:ALEMBIC_CONFIG:conf/alembic.ini}
```
Comment (marcospri) [inline tox.ini:86]: From other projects.

**Review by seanh — APPROVED** [review]

**Comment by marcospri** [comment]
superseded by https://github.com/hypothesis/h/pull/8130

**Pull request closed without merging**